src/ui/zcl_abapgit_gui_chunk_lib.clas.abap

```diff
@@ -153,6 +158,100 @@ CLASS zcl_abapgit_gui_chunk_lib IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD render_error_message_box.
+
+    CONSTANTS: lc_regex_msgid_and_msgno TYPE string VALUE `(.{5})` ##NO_TEXT.
+
+    DATA:
+      lv_error_text      TYPE string,
+      lv_longtext        TYPE string,
+      lv_msgid_and_msgno TYPE string,
+      lv_program_name    TYPE syrepid.
+
+
+    CREATE OBJECT ro_html.
+
+    lv_error_text = ix_error->get_text( ).
+    lv_longtext = ix_error->get_longtext( abap_true ).
+
+    REPLACE ALL OCCURRENCES OF cl_abap_char_utilities=>newline
+            IN lv_longtext
+            WITH '<br>'.
+
+    FIND FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+         IN lv_longtext
+         SUBMATCHES lv_msgid_and_msgno.
+
+    IF sy-subrc = 0.
+      REPLACE FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+              IN lv_longtext
+              WITH ``.
+    ENDIF.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-cause }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-system_response }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-what_to_do }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-sys_admin }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    ro_html->add( |<div id="message" class="message-panel">| ).
+    ro_html->add( |{ lv_error_text }| ).
+    ro_html->add( |<div class="float-right">| ).
+
+    ro_html->add_a(
+        iv_txt   = `&#x274c;`
+        iv_act   = `toggleDisplay('message')`
+        iv_class = `close-btn`
+        iv_typ   = zif_abapgit_html=>c_action_type-onclick ).
+
+    ro_html->add( |</div>| ).
+
+    ro_html->add( |<div class="float-right message-panel-commands">| ).
+
+    IF lv_msgid_and_msgno IS NOT INITIAL.
+      ro_html->add_a(
+          iv_txt = lv_msgid_and_msgno
```

Comment:
Imho, I propose keep "message" text near 00001 - makes it clearer for a user (btw why 5 digits ? :), in fact is the number needed at all ? Having that a direct link is there. Maybe just `message` ?

Reply:
5 digits: ah ! it is message class ! stupid me  ! :)